src/plugins/gmx/gmxplugin.cpp

```diff
@@ -110,16 +110,69 @@ bool GmxPlugin::write(const Map *map, const QString &fileName)
     writeProperty(stream, map, "speed", 30);
     writeProperty(stream, map, "persistent", false);
     writeProperty(stream, map, "clearDisplayBuffer", true);
+    writeProperty(stream, map, "clearViewBackground", false);
+    writeProperty(stream, map, "enableViews", false);
 
     stream.writeTextElement("isometric", toString(map->orientation() == Map::Orientation::Isometric));
 
+    // Write out views
+    // Last view in Object layer is the first view in the room
+    stream.writeStartElement("views");
+    LayerIterator iterator(map);
+    int viewCount = 0;
+    while (const Layer *layer = iterator.next()) {
+
+        if (layer->layerType() != Layer::ObjectGroupType)
+            continue;
+
+        const ObjectGroup *objectLayer = static_cast<const ObjectGroup*>(layer);
+
+          for (const MapObject *object : objectLayer->objects()) {
+            const QString type = effectiveObjectType(object);
+            if (type.isEmpty())
+                continue;
+            if (type.operator !=("view"))
+                continue;
+
+            // GM only has 8 views so drop anything more than that
+            if (viewCount > 7)
+                break;
+
+            viewCount++;
+            stream.writeStartElement("view");
+
+            stream.writeAttribute("visible", toString(object->isVisible()));
+            stream.writeAttribute("objName", QString(optionalProperty(object, "objName", QString(""))));
```

Comment:
Please construct an empty string with `QString()` instead of `QString("")`. It is less verbose and more efficient. :-)